alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultAccessTokenSupplier.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.alloydb;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+
+public class DefaultAccessTokenSupplier implements AccessTokenSupplier {
+
+  private final CredentialFactory credentialFactory;
+
+  /**
+   * Creates an instance with default retry settings.
+   *
+   * @param tokenSource the token source that produces auth tokens.
+   */
+  DefaultAccessTokenSupplier(CredentialFactory tokenSource) {
+    this.credentialFactory = tokenSource;
+  }
+
+  Optional<AccessToken> get() throws IOException {
```

Comment:
Could we inline this method into getTokenValue() and avoid the use of optionals?

Reply:
Updated.